src/openms/include/OpenMS/ANALYSIS/ID/BasicProteinInferenceAlgorithm.h

```diff
@@ -0,0 +1,79 @@
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Julianus Pfeuffer $
+// $Authors: Julianus Pfeuffer $
+// --------------------------------------------------------------------------
+#pragma once
+
+#include <OpenMS/DATASTRUCTURES/DefaultParamHandler.h>
+#include <OpenMS/CONCEPT/ProgressLogger.h>
+#include <OpenMS/METADATA/ProteinIdentification.h>
+#include <OpenMS/METADATA/PeptideHit.h>
+
+namespace OpenMS
+{
+  /// \brief Algorithm class that implements simple protein inference by aggregation of peptide scores
+  /// It has multiple parameter options like the aggregation method, when to distinguish peptidoforms,
+  /// and if you want to use shared peptides.
+  class OPENMS_DLLAPI BasicProteinInferenceAlgorithm :
+    public DefaultParamHandler,
+    public ProgressLogger
+  {
+    public:
+
+    enum AggregationMethod
+    {
+      PROD, SUM, MAXIMUM
+    };
+
+    /// Default constructor
+    BasicProteinInferenceAlgorithm();
+
+    /// main method of BasicProteinInferenceAlgorithm
+    /// inputs are not const, since it will get annotated with results
+    /// annotation of protein groups is currently only possible for a single protein ID run
+    void run(std::vector<PeptideIdentification> &pep_ids, std::vector<ProteinIdentification> &prot_ids) const;
+    void run(std::vector<PeptideIdentification> &pep_ids, ProteinIdentification &prot_id) const;
+
+  private:
+    void processRun_(
```

Comment:
are the different bools etc. set depending on the default param handler? is the documentation of default parameters generated and accessible in the class docu? Might otherwise be hard to properly use the class

Reply:
I moved the reading of the bools from the param object into the function. Described the params in the doxygen docu.